src/oaklib/implementations/sqldb/sql_implementation.py

```diff
@@ -1065,8 +1065,58 @@ def _rbox_relationships(
     def relationships_metadata(
         self, relationships: Iterable[RELATIONSHIP], **kwargs
     ) -> Iterator[Tuple[RELATIONSHIP, List[Tuple[PRED_CURIE, Any]]]]:
-        for rel in relationships:
-            anns = [(ann.predicate, ann.object) for ann in self._axiom_annotations(*rel)]
+        """For given relationships, returns axiom annotation metadata"""
+        # Fetch all annotations
+        anns: List[om.Annotation] = self._axiom_annotations_all()
```

Comment:
I think you are making assumptions about the distributions of numbers of these. Consider a version of omop where every axiom has an annotation

Reply:
By distribution, do you rather mean, "the potentially high number of annotations (returned)", as you query in [this comment](https://github.com/INCATools/ontology-access-kit/pull/659#discussion_r1337998760)?:
> what if n_relationships > 1m?

If that's what you mean, let's continue discussion in that thread and mark this one resolved. If you mean something else though, can you clarify?